references/validationFunctions/common/drugDatabase.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2022 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of the GNU Affero General Public License v3.0.
+ * You should have received a copy of the GNU Affero General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY                           
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES                          
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT                           
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,                                
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED                          
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;                               
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER                              
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN                         
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *  
+ *
+ */
+
+/**
+ * Requirement to submit either drug_rxnormcui, drug_name or drug_database, drug_id, drug_term.
+ */
+
+const validation = () => 
+  (function validate(inputs) {
+    const {$row, $name, $field} = inputs;
+    let result = {valid: true, message: "Ok"};
+
+    // Extract related fields from the row
+    const drug_rxnormcui = $row.drug_rxnormcui;
+    const drug_name = $row.drug_name;
+    const drug_database = $row.drug_database;
+    const drug_id = $row.drug_id;
+    const drug_term = $row.drug_term;
+      
+    // checks for a string just consisting of whitespace
+    const checkforEmpty = (entry) => {return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'))};
+    
+    // Validate based on the field name
+    switch ($name) {
+        case 'drug_rxnormcui':
+            // If drug_rxnormcui is provided
+            if (!checkforEmpty($field) && $field) {
+              if (checkforEmpty(drug_name) || !drug_name) {
+                  result = {
+                    valid: false,
+                    message: `drug_name is required when drug_rxnormcui is provided.`
+                  };
+              }
+              if (!checkforEmpty(drug_database) && drug_database || !checkforEmpty(drug_id) && drug_id || !checkforEmpty(drug_term) && drug_term  ) {
```

Comment:
I'd look at defining these `!checkforEmpty(x) && x` statements as variables rather than repeating them later 
Will read better and can trace potential issues easier

for example: `const isRxNormEmpty = checkforEmpty(drug_rxnormcui) || !drug_rxnormcui`

You can also handle the opposite case: `!isRxNormEmpty` replaces wherever you use `!checkforEmpty(drug_rxnormcui) || drug_rxnormcui`

Reply:
` if ( !isRxNormEmpty || !isDrugIdEmpty || !isDrugTermEmpty )`

Reply:
that's very good idea. thanks Dan!